contracts/Marketplace2.sol

```diff
@@ -0,0 +1,419 @@
+pragma solidity ^0.4.22;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./PurchaseListener.sol";
+import "./Ownable.sol";
+
+
+contract IMarketplace {
+    enum ProductState {
+        NotDeployed,                // non-existent or deleted
+        Deployed                    // created or redeployed
+    }
+
+    enum Currency {
+        DATA,                       // "token wei" (10^-18 DATA)
+        USD                         // attodollars (10^-18 USD)
+    }
+
+    function getProduct(bytes32 id) public view returns (string name, address owner, address beneficiary, uint pricePerSecond, Currency currency, uint minimumSubscriptionSeconds, ProductState state) {}
+    function getSubscription(bytes32 productId, address subscriber) public view returns (bool isValid, uint endTimestamp) {}
+}
+
+/**
+ * @title Streamr Marketplace
+ * @dev note about numbers:
+ *   All prices and exchange rates are in "decimal fixed-point", that is, scaled by 10^18, like ETH vs wei.
+ *  Seconds are integers as usual.
+ *
+ * Next version TODO:
```

Comment:
These TODOs are a leftover of the Marketplace v1 code? Can be cleaned up. Some of them have been already  implemented and can be removed (Bancor, buyFor, grant), and any remaining ones should be filed as JIRA tickets. cc @jtakalai

Reply:
ok I'll remove

Reply:
What is goal of [EIP-165](https://streamr.atlassian.net/browse/MAR-667)? The ticket has no description